### PR TITLE
Fix jaeger labels selector

### PR DESCRIPTION
**Pull request by michaellzc**
We were rendering duplicated labels `app.kubernetes.io/component` and `app.kubernetes.io/name` in jaeger `deploy` and `svc`. This is causing problems when piping rendered output to `kustomize`. 

```
stomize-chart/kustomize. error output:
Error: received error yaml: unmarshal errors:
  line 29: mapping key "app.kubernetes.io/name" already defined at line 27 for the following resource:
```

This PR 

- cleans up all these duplicated labels.
- enforce consistency to some extent without introducing any breaking change
  - use `app.kubernetes.io/component: all-in-one` and `app.kubernetes.io/name: jaeger` as much as possible. They are the value of `matchLabels` right now and we can't change them, so we might as well use them as the standard for all jaeger components.

See below comment for the latest diff

### Test plan

Deploy the chart from the last release

```sh
helm upgrade --install -n sourcegraph -f ./override.yaml sourcegraph sourcegraph/sourcegraph
```

Inspect the diff, jaeger deployment and services should no longer contain duplicate labels.

```sh
helm diff -n sourcegraph -f ./override.yaml sourcegraph charts/sourcegraph/.
```

Deploy the changes make sure it's non-breaking

```sh
helm upgrade --install -n sourcegraph -f ./override.yaml sourcegraph charts/sourcegraph/.
```

Of course, verify tracing still works

### CODE REVIEW

**Comment by michaellzc** [comment]
Current dependencies on/for this PR:
* main
  * **PR #62** <a href="https://app.graphite.dev/github/pr/sourcegraph/deploy-sourcegraph-helm/62" target="_blank">Graphite</a>  (this PR)
    * **PR #63** <a href="https://app.graphite.dev/github/pr/sourcegraph/deploy-sourcegraph-helm/63" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).

File: charts/sourcegraph/templates/jaeger/jaeger-collector.Service.yaml
Comment (michaellzc) [reply charts/sourcegraph/templates/jaeger/jaeger-collector.Service.yaml:11]: @caugustus-sourcegraph 

I ended up putting my thoughts into a mini RFC https://docs.google.com/document/d/1chs2YaI4pCg9meLnt6WFbdj2nxCY_r1U8TTSKOyPmDA/edit

**Comment by michaellzc** [comment]
Updated diff

`jaeger.Deployment.yaml`

```diff
  # Source: sourcegraph/templates/jaeger/jaeger.Deployment.yaml
  apiVersion: apps/v1
  kind: Deployment
  metadata:
    name: jaeger
    labels:
      helm.sh/chart: sourcegraph-0.4.0
-     app.kubernetes.io/name: sourcegraph
-     app.kubernetes.io/instance: sourcegraph
+     app.kubernetes.io/name: jager
      app.kubernetes.io/version: "3.37.0"
      app.kubernetes.io/managed-by: Helm
+     app.kubernetes.io/part-of: sourcegraph
      deploy: sourcegraph
-     app.kubernetes.io/component: jaeger
+     app.kubernetes.io/component: all-in-one
      app: jaeger
-     app.kubernetes.io/name: jaeger
  spec:
    replicas: 1
    revisionHistoryLimit: 10
    selector:
      matchLabels:
+       app.kubernetes.io/name: jager
        app: jaeger
-       app.kubernetes.io/name: jaeger
        app.kubernetes.io/component: all-in-one
    strategy:
      type: Recreate
    template:
      metadata:
        annotations:
          prometheus.io/scrape: "true"
          prometheus.io/port: "16686"
          kubectl.kubernetes.io/default-container: jaeger
        labels:
-         app.kubernetes.io/name: sourcegraph
-         app.kubernetes.io/instance: sourcegraph
+         app.kubernetes.io/name: jager
          app: jaeger
          deploy: sourcegraph
-         app.kubernetes.io/name: jaeger
          app.kubernetes.io/component: all-in-one
```

File: charts/sourcegraph/templates/jaeger/jaeger-collector.Service.yaml
Comment (caugustus-sourcegraph) [inline charts/sourcegraph/templates/jaeger/jaeger-collector.Service.yaml:34]: It looks like the `app.kubernetes.io/instance` labeled isn't being attached to the pods - I think this selector and the service selectors will fail?

Comment (michaellzc) [reply charts/sourcegraph/templates/jaeger/jaeger-collector.Service.yaml:34]: the diff is outdated and I just updated the comment https://github.com/sourcegraph/deploy-sourcegraph-helm/pull/62#issuecomment-1063412878

https://github.com/sourcegraph/deploy-sourcegraph-helm/pull/62/files#diff-3911ffe87f6289604d636cc83ba2a3df8dab9685b2c07c4e49763eb85a87897bR44

**Review by caugustus-sourcegraph — APPROVED** [review]

**Comment by michaellzc** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://graphite.dev?utm_source=gh).


Job Reference: 7AR5iRB1puWVsiuAXgLi

**Pull request merged**
